src/eduid/satosa/scimapi/accr.py

```diff
@@ -0,0 +1,148 @@
+            minimum_accr = self.lowest_accepted_accr_for_virtual_idp.get(virtual_idp)
+        if minimum_accr:
+            logger.debug(f"Minimum accepted ACCR for {virtual_idp} is: {minimum_accr}.")
+            required_accr_by_virtual_idp, _ = self.split_sorted_accr_by(
+                minimum_accr=minimum_accr, supported_accr=self.supported_accr_sorted_by_prio
+            )
+            logger.debug(
+                f"Replacing requested ACCR: {requested_accr}, with what {virtual_idp} requires: {required_accr_by_virtual_idp}."
+            )
+            accr_to_forward = required_accr_by_virtual_idp
+
+        context.state[Context.KEY_TARGET_AUTHN_CONTEXT_CLASS_REF] = accr_to_forward
+
+        return super().process(context, data)
+
+    @staticmethod
+    def split_sorted_accr_by(minimum_accr: str, supported_accr: list) -> tuple[list, list]:
+        result = []
+        temp_list = []
+        for i in supported_accr:
+            if i == minimum_accr:
+                temp_list.append(i)
+                result.append(temp_list)
+                temp_list = []
+            else:
+                temp_list.append(i)
+        result.append(temp_list)
+        return result[0], result[1]
+
+
+class response(ResponseMicroService):
+    """
+    A class to handle and the ACCR response flowing through Satosa.
+    ```yaml
+    module: eduid.satosa.scimapi.accr.response
+    name: accrResponse
+    """
+
+    def __init__(self, config: Mapping[str, Any], internal_attributes: dict[str, Any], *args: Any, **kwargs: Any):
+
+        super().__init__(*args, **kwargs)
+
+    def process(self, context: satosa.context.Context, data: satosa.internal.InternalData) -> ProcessReturnType:
+        received_accr = data.auth_info.auth_class_ref
+        saved_accr = context.state.get("saved_accr")
+        internal_accr_rewrite_map = context.state.get("internal_accr_rewrite_map")
+        supported_accr_sorted_by_prio = context.state.get("supported_accr_sorted_by_prio")
+        logger.debug(f"Received ACCR from IdP: {received_accr}")
+        logger.debug(f"Saved (requested) ACCR from state: {saved_accr}")
+
+        if internal_accr_rewrite_map and received_accr:
+            for origin, rewrite in internal_accr_rewrite_map.items():
+                if received_accr == rewrite:
+                    logger.debug(f"Rewriting ACCR {received_accr} back to {origin}")
+                    received_accr = origin
+                    break
+
+        if not saved_accr:
+            logger.debug(f"No ACCR in request, setting: {received_accr}")
+            data.auth_info.auth_class_ref = received_accr
+        else:
+            for accr in supported_accr_sorted_by_prio:
+                if accr in saved_accr:
+                    logger.debug(f"Setting ACCR to most priorirtied avaliable value in request: {accr}")
+                    data.auth_info.auth_class_ref = accr
+                    break
+            # XXX exception if no match?
```

Comment:
If you don't do anything here whatever already is in `data.auth_info.auth_class_ref` will be used right?

Reply:
Yes, that is correct.